pkg/client/restclient/request.go

```diff
@@ -517,6 +517,20 @@ func (r *Request) Timeout(d time.Duration) *Request {
 	return r
 }
 
+func (r *Request) encoder() (runtime.Encoder, bool) {
+	negotiated := r.content.NegotiatedSerializer
```

Comment:
I think we need to be doing this during client initialization.  If the client fails to start, I'd rather error there than here.  Do we really expect to be changing codecs mid client?